### PR TITLE
Add new API test case for snapshot create / exec

### PR DESCRIPTION
**Pull request by Yuyupo**
JerryScript-DCO-1.0-Signed-off-by: Daniella Barsony bella@inf.u-szeged.hu

### CODE REVIEW

File: tests/unit-core/test-snapshot.c [hunk 1]
```diff
@@ -79,6 +79,59 @@ static void test_function_snapshot (void)
...
+
+static void test_function_arguments_snapshot (void)
+{
+  if (jerry_is_feature_enabled (JERRY_FEATURE_SNAPSHOT_SAVE)
+      && jerry_is_feature_enabled (JERRY_FEATURE_SNAPSHOT_EXEC))
+  {
+    static uint32_t global_arguments_snapshot_buffer[SNAPSHOT_BUFFER_SIZE];
+    static uint32_t eval_arguments_snapshot_buffer[SNAPSHOT_BUFFER_SIZE];
+    const char *code_to_snapshot_p =
+    "function f(a,b,c) { arguments[0]++; arguments[1]++; arguments[2]++; return a + b + c; } f(3,4,5);";
```
Comment (zherczeg) [inline tests/unit-core/test-snapshot.c]: Correct style:

```
const char *code_to_snapshot_p = ("function f(a,b,c) {"
                                  "  arguments[0]++;"
                                  "  arguments[1]++;"
                                  "  arguments[2]++;"
                                  "  return a + b + c;"
                                  "}"
                                  "f(3,4,5);");

File: tests/unit-core/test-snapshot.c [hunk 2]
```diff
@@ -79,6 +79,59 @@ static void test_function_snapshot (void)
...
+  TEST_ASSERT (!jerry_value_has_error_flag (res));
+  TEST_ASSERT (jerry_value_is_number (res));
+  double raw_value = jerry_get_number_value (res);
+  TEST_ASSERT (raw_value == 15);
+  jerry_release_value (res);
+
+  jerry_cleanup ();
+} /* arguments_test_exec_snapshot */
+
+
```
Comment (zherczeg) [inline tests/unit-core/test-snapshot.c]: Single newline.

**Comment by Yuyupo** [comment]
All changes finished. @zherczeg

File: tests/unit-core/test-snapshot.c [hunk 3]
```diff
@@ -79,6 +79,63 @@ static void test_function_snapshot (void)
...
+    TEST_ASSERT (global_mode_snapshot_size != 0);
+    jerry_cleanup ();
+
+    jerry_init (JERRY_INIT_EMPTY);
+    size_t eval_mode_snapshot_size = jerry_parse_and_save_snapshot ((jerry_char_t *) code_to_snapshot_p,
+                                                                      strlen (code_to_snapshot_p),
+                                                                      false,
+                                                                      false,
+                                                                      eval_arguments_snapshot_buffer,
+                                                                      SNAPSHOT_BUFFER_SIZE);
```
Comment (zherczeg) [inline tests/unit-core/test-snapshot.c]: These lines are incorrectly aligned. Please remove 2 spaces before each line.

**Pull request merged**